app/Http/Controllers/CommentsController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Comment;
+use App\Http\Requests\CommentRequest;
+use Illuminate\Http\Request;
+
+class CommentsController extends Controller
+{
+    public function __construct()
+    {
+        // You need to be logged in and have admin rights to access
+        $this->middleware('auth');
+    }
+
+    public function edit(Comment $comment, Request $request)
+    {
+        $origin = $request->query("origin", "/");
+        return view("comments.edit", compact('comment', 'origin'));
+    }
+
+    public function update(Comment $comment, CommentRequest $request)
+    {
+        if ($request->get("is_secret") && !\Auth::user()->is_admin) {
+            abort(403, 'Cannot create secret comments as non-admin');
+        }
+        $comment->update($request->all());
+
+        return redirect($request->query("origin", "/"));
+    }
+
+    public function create(Request $request)
+    {
+        $type = $request->query("type");
+        $key = $request->query("key");
+
+        if (!$type || !$key) {
+            abort(400, 'Invalid input, expected entity type and entity key.');
+        }
+
+        $origin = $request->query("origin", "/");
+        return view("comments.create", compact('comment', 'origin', 'type', 'key'));
```

Comment:
Waar zou de variabele `comment` in de compact array vandaan moeten komen?

Reply:
Nergens, dit was duidelijk het betere copy pasta werk